diff --git a/node_modules/@shipt/segmented-arc-for-react-native/src/SegmentedArc.js b/node_modules/@shipt/segmented-arc-for-react-native/src/SegmentedArc.js
index 2006544..92fd2b5 100644
--- a/node_modules/@shipt/segmented-arc-for-react-native/src/SegmentedArc.js
+++ b/node_modules/@shipt/segmented-arc-for-react-native/src/SegmentedArc.js
@@ -129,7 +129,8 @@ export const SegmentedArc = ({
 
   return (
     <View style={styles.container} testID="container">
-      <Svg width={svgWidth} height={svgHeight}>
+      <Svg width={svgWidth} height={svgHeight} viewBox='0 0 90 90'>
+        
         <SegmentedArcContext.Provider
           value={{
             margin,
diff --git a/node_modules/@shipt/segmented-arc-for-react-native/src/components/Cap.js b/node_modules/@shipt/segmented-arc-for-react-native/src/components/Cap.js
index 6ab2619..e891059 100644
--- a/node_modules/@shipt/segmented-arc-for-react-native/src/components/Cap.js
+++ b/node_modules/@shipt/segmented-arc-for-react-native/src/components/Cap.js
@@ -43,7 +43,7 @@ export const Cap = () => {
 
   return (
     <G>
-      <ArcCap r="10" fill={capOuterColor} ref={outerCircleRef} {...initialCoordinates} />
+      <ArcCap r="8" fill={capOuterColor} ref={outerCircleRef} {...initialCoordinates} />
       <ArcCap r="6" fill={capInnerColor} ref={innerCircleRef} {...initialCoordinates} />
     </G>
   );
