diff --git a/node_modules/native-base/lib/commonjs/theme/styled-system.js b/node_modules/native-base/lib/commonjs/theme/styled-system.js
index c7c18f3..0dc1c01 100644
--- a/node_modules/native-base/lib/commonjs/theme/styled-system.js
+++ b/node_modules/native-base/lib/commonjs/theme/styled-system.js
@@ -128,6 +128,10 @@ const flexbox = {
   justifyContent: true,
   flexWrap: true,
   flexDirection: true,
+  direction: {
+    property: 'flexDirection',
+    scale: 'flexDirection'
+  },
   flexDir: {
     property: 'flexDirection',
     scale: 'flexDirection'
diff --git a/node_modules/native-base/lib/module/theme/styled-system.js b/node_modules/native-base/lib/module/theme/styled-system.js
index 8c01971..d02115f 100644
--- a/node_modules/native-base/lib/module/theme/styled-system.js
+++ b/node_modules/native-base/lib/module/theme/styled-system.js
@@ -111,6 +111,10 @@ export const flexbox = {
   justifyContent: true,
   flexWrap: true,
   flexDirection: true,
+  direction: {
+    property: 'flexDirection',
+    scale: 'flexDirection'
+  },
   flexDir: {
     property: 'flexDirection',
     scale: 'flexDirection'
diff --git a/node_modules/native-base/src/hooks/useKeyboardDismissable.ts b/node_modules/native-base/src/hooks/useKeyboardDismissable.ts
index f3cfa44..953688f 100644
--- a/node_modules/native-base/src/hooks/useKeyboardDismissable.ts
+++ b/node_modules/native-base/src/hooks/useKeyboardDismissable.ts
@@ -48,12 +48,17 @@ export function useBackHandler({ enabled, callback }: IParams) {
       callback();
       return true;
     };
+    let subscription: any;
     if (enabled) {
-      BackHandler.addEventListener('hardwareBackPress', backHandler);
-    } else {
-      BackHandler.removeEventListener('hardwareBackPress', backHandler);
+      subscription = BackHandler.addEventListener(
+    'hardwareBackPress',
+        backHandler
+      );
     }
-    return () =>
-      BackHandler.removeEventListener('hardwareBackPress', backHandler);
+    return () => {
+      if (subscription && typeof subscription.remove === 'function') {
+        subscription.remove();
+      }
+    };
   }, [enabled, callback]);
 }
diff --git a/node_modules/native-base/src/theme/components/input.ts b/node_modules/native-base/src/theme/components/input.ts
index ed3047c..efb63dd 100644
--- a/node_modules/native-base/src/theme/components/input.ts
+++ b/node_modules/native-base/src/theme/components/input.ts
@@ -17,7 +17,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
       },
     },
     _web: {
-      outlineWidth: '0',
+      outlineWidth: 0,
       overflow: 'auto',
       lineHeight: 'lg', // Todo: Move to _web inside size so that sm and xs don't have this much height
       style: { outline: 'none' },
@@ -47,7 +47,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
       _hover: { borderColor: 'error.600' },
       _stack: {
         style: {
-          outlineWidth: '0',
+          outlineWidth: 0,
           boxShadow: `0 0 0 1px ${error[600]}`,
         },
       },
@@ -60,7 +60,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
         _hover: { borderColor: 'error.600' },
         _stack: {
           style: {
-            outlineWidth: '0',
+            outlineWidth: 0,
             boxShadow: `0 0 0 1px ${error[600]}`,
           },
         },
@@ -79,7 +79,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
       },
       _stack: {
         style: {
-          outlineWidth: '0',
+          outlineWidth: 0,
           boxShadow: `0 0 0 1px ${primary[600]}`,
         },
       },
@@ -96,7 +96,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
         _hover: { borderColor: 'primary.500' },
         _stack: {
           style: {
-            outlineWidth: '0',
+            outlineWidth: 0,
             boxShadow: `0 0 0 1px ${primary[500]}`,
           },
         },
@@ -105,7 +105,7 @@ const baseStyle = (props: InterfaceInputProps & { theme: any }) => {
         borderColor: 'error.500',
         _stack: {
           style: {
-            outlineWidth: '0',
+            outlineWidth: 0,
             boxShadow: `0 0 0 1px ${error[500]}`,
           },
         },
@@ -190,7 +190,7 @@ function unstyledStyle() {
     _stack: {
       _focus: {
         style: {
-          outlineWidth: '0',
+          outlineWidth: 0,
         },
       },
     },
@@ -206,7 +206,7 @@ function underlinedStyle(props: InterfaceInputProps & { theme: any }) {
     _focus: {
       _stack: {
         style: {
-          outlineWidth: '0',
+          outlineWidth: 0,
           boxShadow: `0 1px 0 0 ${primary[600]}`,
         },
       },
@@ -224,7 +224,7 @@ function underlinedStyle(props: InterfaceInputProps & { theme: any }) {
       _focus: {
         _stack: {
           style: {
-            outlineWidth: '0',
+            outlineWidth: 0,
             boxShadow: `0 1px 0 0 ${primary[500]}`,
           },
         },
diff --git a/node_modules/native-base/src/theme/styled-system.ts b/node_modules/native-base/src/theme/styled-system.ts
index 575d972..1aceca9 100644
--- a/node_modules/native-base/src/theme/styled-system.ts
+++ b/node_modules/native-base/src/theme/styled-system.ts
@@ -112,6 +112,10 @@ export const flexbox = {
   justifyContent: true,
   flexWrap: true,
   flexDirection: true,
+  direction: {
+    property: 'flexDirection',
+    scale: 'flexDirection',
+  },
   flexDir: {
     property: 'flexDirection',
     scale: 'flexDirection',
