diff --git a/node_modules/react-native-html-to-pdf/android/src/main/java/android/print/PdfConverter.java b/node_modules/react-native-html-to-pdf/android/src/main/java/android/print/PdfConverter.java
index 5d61c5f..6072d55 100644
--- a/node_modules/react-native-html-to-pdf/android/src/main/java/android/print/PdfConverter.java
+++ b/node_modules/react-native-html-to-pdf/android/src/main/java/android/print/PdfConverter.java
@@ -112,6 +112,8 @@ public class PdfConverter implements Runnable {
         });
         WebSettings settings = mWebView.getSettings();
         settings.setTextZoom(100);
+        settings.setJavaScriptEnabled(true);
+        settings.setJavaScriptCanOpenWindowsAutomatically(true);
         settings.setDefaultTextEncodingName("utf-8");
         settings.setAllowFileAccess(true);
         mWebView.loadDataWithBaseURL(mBaseURL, mHtmlString, "text/HTML", "utf-8", null);
