diff --git a/node_modules/react-native-tor/android/src/main/java/com/reactnativetor/TorModule.kt b/node_modules/react-native-tor/android/src/main/java/com/reactnativetor/TorModule.kt
index df470c9..a32f364 100644
--- a/node_modules/react-native-tor/android/src/main/java/com/reactnativetor/TorModule.kt
+++ b/node_modules/react-native-tor/android/src/main/java/com/reactnativetor/TorModule.kt
@@ -151,7 +151,7 @@ class TorModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMod
       return;
     }
 
-    val param = TaskParam(method, url, jsonBody, headers.toHashMap())
+    val param = TaskParam(method, url, jsonBody, headers.toHashMap() as HashMap<String, Any>?)
     executorService.execute {
       try {
         val task = TorBridgeRequest(promise, _client!!, param);
