diff --git a/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm b/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
index 74ebc31..8ebdeac 100644
--- a/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
@@ -10,6 +10,7 @@
 #import <Foundation/NSMapTable.h>
 #import <React/RCTAssert.h>
 #import <React/RCTConstants.h>
+#import <React/RCTLog.h>
 
 #import <React/RCTImageComponentView.h>
 #import <React/RCTParagraphComponentView.h>
@@ -111,8 +112,11 @@ - (void)_enqueueComponentViewWithComponentHandle:(ComponentHandle)componentHandl
     return;
   }
 
-  RCTAssert(
-      componentViewDescriptor.view.superview == nil, @"RCTComponentViewRegistry: Attempt to recycle a mounted view.");
+  if (componentViewDescriptor.view.superview != nil) {
+    RCTLogWarn(@"RCTComponentViewRegistry: Skipping recycle for mounted view %@", componentViewDescriptor.view);
+    return;
+  }
+
   [componentViewDescriptor.view prepareForRecycle];
 
   recycledViews.push_back(componentViewDescriptor);
