diff --git a/node_modules/react-native-document-picker/android/src/main/java/com/reactnativedocumentpicker/RNDocumentPickerModule.java b/node_modules/react-native-document-picker/android/src/main/java/com/reactnativedocumentpicker/RNDocumentPickerModule.java
index 5001525..02a467f 100644
--- a/node_modules/react-native-document-picker/android/src/main/java/com/reactnativedocumentpicker/RNDocumentPickerModule.java
+++ b/node_modules/react-native-document-picker/android/src/main/java/com/reactnativedocumentpicker/RNDocumentPickerModule.java
@@ -18,7 +18,6 @@ import androidx.annotation.NonNull;
 import com.facebook.react.bridge.ActivityEventListener;
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.BaseActivityEventListener;
-import com.facebook.react.bridge.GuardedResultAsyncTask;
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContext;
@@ -223,20 +222,21 @@ public class RNDocumentPickerModule extends NativeDocumentPickerSpec {
         return;
       }
 
-      new ProcessDataTask(getReactApplicationContext(), uris, copyTo, promise).execute();
+      new Thread(new ProcessDataTask(getReactApplicationContext(), uris, copyTo, promise)).start();
     } catch (Exception e) {
       sendError(E_UNEXPECTED_EXCEPTION, e.getLocalizedMessage(), e);
     }
   }
 
-  private static class ProcessDataTask extends GuardedResultAsyncTask<ReadableArray> {
+  private static class ProcessDataTask implements Runnable {
     private final WeakReference<Context> weakContext;
+    private final ReactContext reactContext;
     private final List<Uri> uris;
     private final String copyTo;
     private final Promise promise;
 
     protected ProcessDataTask(ReactContext reactContext, List<Uri> uris, String copyTo, Promise promise) {
-      super(reactContext.getExceptionHandler());
+      this.reactContext = reactContext;
       this.weakContext = new WeakReference<>(reactContext.getApplicationContext());
       this.uris = uris;
       this.copyTo = copyTo;
@@ -244,17 +244,17 @@ public class RNDocumentPickerModule extends NativeDocumentPickerSpec {
     }
 
     @Override
-    protected ReadableArray doInBackgroundGuarded() {
-      WritableArray results = Arguments.createArray();
-      for (Uri uri : uris) {
-        results.pushMap(getMetadata(uri));
+    public void run() {
+      try {
+        final WritableArray results = Arguments.createArray();
+        for (Uri uri : uris) {
+          results.pushMap(getMetadata(uri));
+        }
+        reactContext.runOnUiQueueThread(() -> promise.resolve(results));
+      } catch (Exception e) {
+        reactContext.runOnUiQueueThread(
+            () -> promise.reject(E_UNEXPECTED_EXCEPTION, e.getLocalizedMessage(), e));
       }
-      return results;
-    }
-
-    @Override
-    protected void onPostExecuteGuarded(ReadableArray readableArray) {
-      promise.resolve(readableArray);
     }
 
     private WritableMap getMetadata(Uri uri) {
